lte/gateway/deploy/agw_install.sh

```diff
@@ -70,24 +77,64 @@ if [ "$KVERS" != "4.9.0-9-amd64" ]; then
   NEED_REBOOT=1
 fi
 
+# configure environment variable defaults needed for ansible
+ANSIBLE_VARS="PACKAGE_LOCATION=/tmp"
+if [ -n "${REPO_HOST}" ]; then
+    if [ -z "${REPO_PROTO}" ]; then
+        REPO_PROTO=http
+    fi
+    if [ -z "${REPO_DIST}" ]; then
+        REPO_DIST=stretch-stable
+    fi
+    if [ -z "${REPO_COMPONENT}" ]; then
+        REPO_COMPONENT=main
+    fi
+    # configure pkgrepo location
+    ANSIBLE_VARS="ovs_pkgrepo_proto=${REPO_PROTO} ovs_pkgrepo_host=${REPO_HOST} ovs_pkgrepo_path=${REPO_PATH} ${ANSIBLE_VARS}"
+
+    # configure pkgrepo distribution
+    ANSIBLE_VARS="ovs_pkgrepo_dist=${REPO_DIST} ovs_pkgrepo_component=${REPO_COMPONENT} ${ANSIBLE_VARS}"
+
+    # configure pkgrepo gpg key
+    ANSIBLE_VARS="ovs_pkgrepo_key=${REPO_KEY} ${ANSIBLE_VARS}"
+    if [ -z "${REPO_KEY_FINGERPRINT}" ]; then
+        ANSIBLE_VARS="ovs_pkgrepo_key_fingerprint=${REPO_KEY_FINGERPRINT} ${ANSIBLE_VARS}"
+    fi
+fi
+
+if [ "${REPO_PROTO}" == 'https' ]; then
+    echo "Ensure HTTPS apt transport method is installed"
+    apt install -y apt-transport-https
+fi
+
 if [ $NEED_REBOOT = 1 ]; then
   echo "Will reboot in a few seconds, loading a boot script in order to install magma"
   if [ ! -f "$AGW_SCRIPT_PATH" ]; then
-    wget --no-cache -O $AGW_SCRIPT_PATH "https://raw.githubusercontent.com/facebookincubator/magma/$MAGMA_VERSION/lte/gateway/deploy/agw_install.sh"
+      cp "$(realpath $0)" "${AGW_SCRIPT_PATH}"
   fi
-  echo "[Unit]
+  cat <<EOF > $AGW_INSTALL_CONFIG
```

Comment:
just curiosity why change the `echo` to a `cat`?

Reply:
to allow HEREDOC syntax cleanly